src/core/transactions.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * SKALE Metaport
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file transactions.ts
+ * @copyright SKALE Labs 2023-Present
+ */
+
+import { type TransactionResponse } from 'ethers';
+import { TRANSFER_ERROR_MSG } from './constants'
+import { TxResponse } from './interfaces'
+
+
+export async function sendTransaction(func: any, args: any[]): Promise<TxResponse> {
+  try {
+    const response: TransactionResponse = await func(...args)
+    await response.wait()
+    return { status: true, err: undefined, response: response }
+  } catch (err) {
+    console.error(err)
```

Comment:
Ideally should not print errors in production unless via different debug channel.